File: dev-docs/snippets/events/workflow-status.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when a workflow run reaches a terminal status (completed/failed/terminated).
export function workflowStatus(client: JsBaoClient) {
  // #region example
  client.on("workflowStatus", (payload) => {
    console.log(payload.workflowKey, "->", payload.status);
  });
  // #endregion example
}

File: dev-docs/snippets/events/workflow-status.swift
import JsBaoClient

// Fires when a workflow run reaches a terminal status (completed/failed/terminated).
func workflowStatus(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.workflowStatus) { (event: WorkflowStatusEvent) in
    print(event.workflowKey, "->", event.status)
  }
  // #endregion example
  _ = sub
}
